queries.sql

```diff
@@ -21,5 +21,65 @@ SELECT * FROM animals WHERE neutered = true;
 /*Find all animals not named Gabumon*/
 SELECT * FROM animals WHERE NOT name LIKE 'Gabumon';
 
-/*Find all animals with a weigh between 10.4kg and 17.3kg (including the animals with the weights that equals precisely 10.4kg or 17.3kg)*/
-SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
\ No newline at end of file
+/*Find all animals with a weigh between 10.4kg and 17.3kg 
+(including the animals with the weights that equals precisely 10.4kg or 17.3kg)*/
+SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+/*Inside a transaction UPDATE the animals table by SETting the species column to unspecified. Verify that change was made. 
+Then roll back the change and verify that species columns went back to the state before transaction.*/
+BEGIN;
+UPDATE animals SET species = 'unspecified';
+SELECT * FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+
+/*Inside a transaction:
+Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
+Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
+Commit the transaction.
+Verify that change was made and persists after commit.*/
+BEGIN;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species IS NLL;
+COMMIT;
+SELECT * FROM animals;
+
+/*Now, take a deep breath and... Inside a transaction DELETE all records in the animals table, then roll back the transaction.
+After the roll back verify if all records in the animals table still exist. After that you can start breath as usual ;)*/
+BEGIN;
+DELETE FROM animals;
+SELECT * FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+
+/*Inside a transaction:
+Delete all animals born after Jan 1st, 2022.
+Create a savepoint for the transaction.
+Update all animals' weight to be their weight multiplied by -1.
+Rollback to the savepoint
+Update all animals' weights that are negative to be their weight multiplied by -1.
+Commit transaction*/
+BEGIN;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT savepoint_1;
+UPDATE animals SET weight_kg=weight_kg*-1 WHERE weight_kg < 0;
+COMMIT;
+SELECT * FROM animals;
```

Comment:
- [ ] I think you are missing one step here. According to the requirements, after creating the savepoint, you should update all animals' weights to be their weight multiplied by -1 and then you should roll back to the save point, which you are not doing right now, so please fix this issue.

Reply:
Thanks for catching that as well, I did do it correctly too in my console but forgot to add it, I'll be more careful In the future.